### PR TITLE
feat: TGeo python binding 

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds some first python bindings for the `TGeo` plugin.

The added functionality is mainly for testing reason, it allows to you test whether sensitive elements from a `TGeo` description (or those one picks as sensitive) can be translated into TGeoDetectorElements.

--- END COMMIT MESSAGE ---

## Summary by CodeRabbit

- **New Features**
  - Introduced TGeo functionality in Python bindings for enhanced geometry integration.
  - Added methods for converting GDML files into TGeo detector elements.
  - Implemented a new conditional build option for TGeo plugin integration.

- **Bug Fixes**
  - Improved error handling in the TGeo conversion process to ensure geometry manager initialization.

- **Documentation**
  - Updated documentation to reflect new methods and classes related to TGeo integration.

### CODE REVIEW

File: Examples/Python/src/TGeo.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  {
+    /// Helper function to test if the automatic geometry conversion works
+    ///
+    /// @param rootFileName is the name of the GDML file
+    /// @param sensitiveMatches is a list of strings to match sensitive volumes
+    /// @param localAxes is the TGeo->ACTS axis conversion convention
+    /// @param convertMaterial is a flag to convert the material
+    tgeo.def("convertToElements",
```
Comment (paulgessinger) [inline Examples/Python/src/TGeo.cpp:47]: If this is an internal function, maybe `_convertToElements`?

Comment (asalzburger) [reply Examples/Python/src/TGeo.cpp:47]: Ok, are we doing this consistently now? 

I am happy to change, I've also added some information why this method is useful.

Comment (asalzburger) [reply Examples/Python/src/TGeo.cpp:47]: Ok, understood, probably we should make some sort of a policy for that then. What I want to achieve here is for people who have a TGeo object in hand to test whether the conversion into detector elements works or not.

Comment (asalzburger) [reply Examples/Python/src/TGeo.cpp]: Fixed.

**Comment by asalzburger** [comment]
@paulgessinger ping ping

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [8eba09e4ca2ecd6ab34330788f21e0f1c04e976d]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/9382852)!

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685090)||
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685089)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685088)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685087)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685086)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685085)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685084)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflowHgtd](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685083)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685082)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685081)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685080)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685079)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685078)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685077)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685076)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685075)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685074)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685073)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685072)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685071)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685070)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685069)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685068)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685067)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685066)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685065)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685064)||
|:green_circle: | [test_ActsGx2fRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685063)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685062)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685061)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685060)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47685059)||

**Pull request merged**